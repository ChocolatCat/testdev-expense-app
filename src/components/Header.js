import React from 'react';
import {NavLink} from 'react-router-dom';

//React Router v6 -> activeClassName changes to that if statement

//<NavLink to="/help" className={({isActive}) => (isActive ? 'is-active' : 'inactive')}>I need help!</NavLink>
const Header = () => (
    <header>
        <h1>Expense App!</h1>
        <NavLink to="/" className={({isActive}) => (isActive ? 'is-active' : 'inactive')}>Home</NavLink>
        <NavLink to="/create" className={({isActive}) => (isActive ? 'is-active' : 'inactive')}>Create expense</NavLink>
    </header>
);

export default Header;