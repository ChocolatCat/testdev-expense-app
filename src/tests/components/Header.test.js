/** @jest-environment jsdom */
import React from 'react';
import Header from '../../components/Header';
import {render, screen} from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom';

//ALL ELEMENTS USING AN ELEMENT FORM REACT ROUTER NEED A ROUTER

it('renders Header without crashing and find the title', () => {
    const component = (
        <BrowserRouter>
            <Header />
        </BrowserRouter>
    );
    render(component);
    expect(screen.getByText("Expense App!")).toBeInTheDocument();
});

it('renders as snapshot', () => {
    const {asFragment} = render(
        <BrowserRouter>
            <Header />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});