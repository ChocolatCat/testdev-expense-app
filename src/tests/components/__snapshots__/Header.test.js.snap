// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Expense App!
    </h1>
    <a
      aria-current="page"
      class="is-active"
      href="/"
    >
      Home
    </a>
    <a
      class="inactive"
      href="/create"
    >
      Create expense
    </a>
  </header>
</DocumentFragment>
`;
